docs: write about composite index strings limitation

lasers opened the pull request:
This adds information about composite index strings limitation. https://github.com/ultrabug/py3status/issues/1516

lasers commented:
I have a thought to remove all button. Stick with commands as intent. Removes error prone too.

ultrabug commented:
@tobes do you have any proposal?

tobes commented:
> @tobes do you have any proposal?

My approach would see dropping of the numerical indexes.  I don't think they are really a used feature and it adds complications and no longer really makes sense.

ultrabug commented:
Do we want to settle for 3.14 ?

tobes commented:
I'd like to remove the 'mis-feature'

ultrabug commented:
> I'd like to remove the 'mis-feature'

then open a PR for this maybe?

lasers commented:
I looked into this. It seems like `i3bar` discards keys not in this dict:
```python
{
 "full_text": "E: 10.0.0.1 (1000 Mbit/s)",
 "short_text": "10.0.0.1",
 "color": "#00ff00",
 "background": "#1c1c1c",
 "border": "#ee0000",
 "min_width": 300,
 "align": "right",
 "urgent": false,
 "name": "ethernet",
 "instance": "eth0",
 "separator": true,
 "separator_block_width": 9
}
```
I tried `'index'` and `'_index'`. Both gets discarded. I think the latter could not be discarded.
>If you want to put in your own entries into a block, prefix the key with an underscore (_). i3bar will ignore all keys it doesn’t understand, and prefixing them with an underscore makes it clear in every script that they are not part of the i3bar protocol.
>**Example:**
>```python
>{
>  "full_text": "E: 10.0.0.1 (1000 Mbit/s)",
>  "_ethernet_vendor": "Intel"
>}
1) We could ask `i3` to see their thoughts on not discarding underscore keys.
2) Put things in `instance` and split them up later (current).
    * Merge this PR and modules stays with `string.split('/')`, dict mapping, etc.

Just a FYI.
